lib/widgets/osm_element_layer/osm_element_layer.dart

```diff
@@ -75,14 +75,21 @@ class _OsmElementLayerState extends State<OsmElementLayer> {
 
   void _handleElementChange(ElementUpdate change) {
     setState(() {
-      if (!change.matches) {
-        _superCluster.remove(change.element);
-      }
-      else if (_superCluster.containsPoint(change.element)) {
-        _superCluster.modifyPointData(change.element, change.element);
+     if (change.action == ElementUpdateAction.clear) {
+        _superCluster.load([]);
       }
       else {
-        _superCluster.insert(change.element);
+          if (change.action == ElementUpdateAction.update) {
+            if (_superCluster.containsPoint(change.element!)) {
+              _superCluster.modifyPointData(change.element!, change.element!);
+            }
+            else {
+              _superCluster.insert(change.element!);
+            }
+          } 
+          else if (change.action == ElementUpdateAction.remove) {
+            _superCluster.remove(change.element!);
+          }
       }
```

Comment:
This can be flattened. Also the indention seems incorrect.

```dart
if (change.action == ElementUpdateAction.clear){
  _superCluster.load([]);
}
else if (change.action == ElementUpdateAction.update) {
  if (_superCluster.containsPoint(change.element!)) {
    _superCluster.modifyPointData(change.element!, change.element!);
  }
  else {
    _superCluster.insert(change.element!);
  }
} 
else if (change.action == ElementUpdateAction.remove){
  _superCluster.remove(change.element!);
}

```